src/core/ledger/currency.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import {
+  ethAddressParser,
+  type EthAddress,
+} from "../../plugins/ethereum/ethAddress";
+import * as C from "../../util/combo";
+
+/**
+ * EvmChainId is represented in number form for all EVM-based chains,
+ * including mainnet (1), and xDai (100)
+ * The reason for this is that ethereum's client configuration utilizes
+ * a number to represent chainId, and this way we can just transpose that
+ * chainId here as a component of the currency Id, since the web3 client will
+ * return a stringified integer when the chainId is requested.
+ */
+export type EvmChainId = string;
+
+export function parseEvmChainId(id: string): EvmChainId {
+  const result = parseInt(id, 10).toString();
+  if (Number.isNaN(result) || result !== id) {
+    throw new Error(`Invalid EVM chainId value: ${id}`);
+  }
+  return id;
+}
+
+export const evmChainParser: C.Parser<EvmChainId> = C.fmap(
+  C.string,
+  parseEvmChainId
+);
+
+declare type EvmId = {
```

Comment:
Maybe `evmToken` would be a better name?

Reply:
Same explanation as above.

Reply:
Not sure I follow. `EvmId` implies that this type is an ID of some sort when in reality it's an object that contains information about some token and the chain it's running on. The EvmId name might be more appropriate if this type was a string in the form of `"evm:chainId:tokenAddress"` and coming out of a parser, but since its an object calling it an ID has a smell to it